Sources/VLCTranscoder.m

```diff
@@ -0,0 +1,142 @@
+    libvlc_media_add_option(p_media, [[NSString stringWithFormat:@"--sub-file=%@", srtPath] UTF8String]);
+    libvlc_media_add_option(p_media, [transcodingOptions UTF8String]);
+
+    _p_mp = libvlc_media_player_new_from_media( p_media );
+    if (_p_mp == NULL) {
+        NSAssert(0, @"_p_mp wasn't allocated");
+        return NO;
+    }
+
+    [self registerObserversForMuxWithPlayer:_p_mp];
+    BOOL canPlay = libvlc_media_player_play( _p_mp ) == 0;
+    if (!canPlay) {
+        NSAssert(0, @"playback failed");
+        [self unregisterObserversForMuxWithPlayer:_p_mp];
+        return NO;
+    }
+    return canPlay;
+}
+
+- (void)registerObserversForMuxWithPlayer:(libvlc_media_player_t *)player
+{
+    __block libvlc_event_manager_t * p_em = libvlc_media_player_event_manager(player);
+    if (!p_em)
+        return;
+    dispatch_sync(_libVLCTranscoderQueue,^{
+        libvlc_event_attach(p_em, libvlc_MediaPlayerPaused,
+                            HandleMuxMediaInstanceStateChanged, (__bridge void *)(self));
+        libvlc_event_attach(p_em, libvlc_MediaPlayerStopped,
+                            HandleMuxMediaInstanceStateChanged, (__bridge void *)(self));
+        libvlc_event_attach(p_em, libvlc_MediaPlayerEncounteredError,
+                            HandleMuxMediaInstanceStateChanged, (__bridge void *)(self));
+    });
+}
+
+- (void)unregisterObserversForMuxWithPlayer:(libvlc_media_player_t *)player
+{
+    libvlc_event_manager_t * p_em = libvlc_media_player_event_manager(player);
+    if (!p_em)
+        return;
+    libvlc_event_detach(p_em, libvlc_MediaPlayerStopped,
+                        HandleMuxMediaInstanceStateChanged, (__bridge void *)(self));
+    libvlc_event_detach(p_em, libvlc_MediaPlayerPaused,
+                        HandleMuxMediaInstanceStateChanged, (__bridge void *)(self));
+    libvlc_event_detach(p_em, libvlc_MediaPlayerEncounteredError,
+                        HandleMuxMediaInstanceStateChanged, (__bridge void *)(self));
+}
+
+- (void)mediaPlayerStateChangeForMux:(NSNumber *)newState
+{
+    if (_p_mp) {
+        [self unregisterObserversForMuxWithPlayer:_p_mp];
+        libvlc_media_player_stop( _p_mp );
+        if (self.delegate && [self.delegate respondsToSelector:@selector(transcoderFinishedSucessfully:)]) {
+            BOOL success = [newState isEqualToNumber: @(VLCMediaPlayerStateError)];
+            [self.delegate transcoderFinishedSucessfully:!success];
+        }
+    }
+}
+
+static void HandleMuxMediaInstanceStateChanged(const libvlc_event_t * event, void * self)
+{
+    VLCMediaPlayerState newState;
+    if (event->type == libvlc_MediaPlayerPaused || event->type == libvlc_MediaPlayerEncounteredError || event->type == libvlc_MediaPlayerStopped) {
```

Comment:
I am having trouble understanding what the purpose of this check here is.

Reply:
yeah it's obsolete looking at it

Reply:
but still the check is obsolete because these are the only things I'm registering for so it will only be one of those three states